cartridge/roles/vshard-storage.lua

```diff
@@ -29,11 +31,33 @@ local function apply_config(conf, _)
 end
 
 local function init()
+    _G_vhsard_backup = rawget(_G, 'vshard')
```

Comment:
Afaics, your considerations don't conflict with the current change. The backup is only saved since `init` to `stop`. I'll add another line to clear it after the role is stopped.